.github/workflows/release.yaml

```diff
@@ -0,0 +1,116 @@
+name: Release
+on:
+  workflow_dispatch:
+    inputs:
+      release:
+        required: false
+        type: boolean
+        default: false
+      release-to-pypi:
+        required: false
+        type: boolean
+        default: false
+    push: # todo: rm push-trigger again after testing (we want release to be triggered manually)
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    uses: ./.github/workflows/build-and-test.yaml
+    with:
+      release: ${{ inputs.release }}
+      release-to-pypi: ${{ inputs.release-to-pypi }}
+
+
+  publish-release-and-bump-commit:
+    runs-on: ubuntu-latest
+    needs:
+      - build
+    steps:
+      - name: install git
+        run: |
+          if ! which git &>/dev/null; then
+            apt-get install git -y
+          fi
+      - uses: actions/checkout@v4
+      - name: Import Release-Commit
+        uses: ./.github/actions/import-commit
+        with:
+          commit-objects: ${{ needs.build.outputs.release-commit-objects }}
+          commit-digest: ${{ needs.build.outputs.release-commit-digest }}
+
+      - name: debug
+        run: |
+          git show
+      - name: Push Release and Bump-Commits
+        id: releasecommit
+        if: ${{ inputs.release }}
+        run: |
+          echo "gha-creds*" >> .git/info/exclude
+          git config --global --add safe.directory $PWD
+          git config user.name 'Gardener-CICD Bot'
+          git config user.email 'gardener.ci.user@gmail.com'
+          effective_version=${{ needs.build.outputs.effective-version }}
+          commit_digest=$(git rev-parse @)
+          tgt_ref="refs/tags/${effective_version}"
+          echo "pushing release-commit ${commit_digest} to ${tgt_ref}"
+          git push origin "@:${tgt_ref}"
+          next_version=${{ needs.version.outputs.next_version }}
+          echo "next version: ${next_version}"
+          git reset --hard @~
+          echo "${next_version}" | .ci/write-version
+          git add .
+          git commit -m"Prepare next Dev-Cycle"
+          git pull --rebase
+          git push origin
+      - name: Retrieve Documentation
+        uses: actions/download-artifact@v4
```

Comment:
This is only required if `inputs.release` is set I suppose?

Reply:
good catch!